extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -64,9 +59,23 @@ class AmpYoutube extends AMP.BaseElement {
         this.element.getAttribute('video-id')),
         'The data-videoid attribute is required for <amp-youtube> %s',
         this.element);
+  }
 
+  /** @override */
+  prerenderCallback() {
     if (!this.getPlaceholder()) {
-      this.buildImagePlaceholder_();
+      // Image load promise.
+      // Cancel this if the layoutCallback promise finished before it did.
+      return this.buildImagePlaceholder_();
```

Comment:
Wouldn't the runtime still manage the `placeholder` regardless where it was added? The runtime calls `togglePlaceholder` inside `firstLayoutComplete` callback, so this is still fine. Although I am happy to the "building" part of this into `buildCallback` - meaning building and appending DOM without `src` - keeping the load in the `prerenderCallback`.

Either way, this would still work as expected where the runtime actually hides the placeholder itself - without the `amp-youtube` worrying about it.

The only reason in `amp-youtube#buildImagePlaceholder_` where we're manually hiding and showing the placeholder is that there was a problem with `object-fit: cover` styling taking a long time to apply for some reason (now that I think about it, it might be because it needs width/height of the element? not sure) - that's why we actually hide the thumbnail initially until it loads to avoid a flash of the thumbnail changing style and until the object-fit takes effect.

This might be a bug in Chrome - might be worth filing a bug for them.